packages/mobile/src/web3/contracts.ts

```diff
@@ -81,10 +82,18 @@ function getWeb3(): Web3 {
   }
 }
 
-export function addLocalAccount(web3Instance: Web3, privateKey: string) {
-  if (!isZeroSyncMode()) {
-    throw new Error('addLocalAccount can only be called in Zero sync mode')
+// Mutates web3 with new provider
+export function switchWeb3ProviderForSyncMode(zeroSync: boolean) {
+  if (zeroSync) {
```

Comment:
Can we get a bit more logging here? Unless you already have it elsewhere where it's called.

Reply:
There's already some logging where it's called but I added logging for each provider switch